CONTRIBUTING.md

```diff
@@ -167,10 +168,24 @@ https://help.github.com/en/github/getting-started-with-github/git-and-github-lea
 - Python [docstrings](http://www.python.org/dev/peps/pep-0257/) need to be in
   [reStructured Text (RST)](https://docutils.sourceforge.io/rst.html) format and
   compatible with [Sphinx](https://www.sphinx-doc.org).
+- String formatting should normally be done using
+  [formatted string literals (f-strings)](https://docs.python.org/3/tutorial/inputoutput.html#formatted-string-literals),
+  except:
+  - when the format string is kept in a separate variable, in which case the
+    [string ``format()`` method](https://docs.python.org/3/tutorial/inputoutput.html#the-string-format-method)
+    should be used;
+  - when [formatting a log message](https://docs.python.org/3/library/logging.html#logging.Logger.debug),
+    in which case the message format string (a
+    [`printf`-style](https://docs.python.org/3/library/stdtypes.html#old-string-formatting)
+    string literal) and its arguments should be passed separately to the logging
+    method.
+    This is a bit more efficient than using f-strings and allows for better log
```

Comment:
I skimmed the post, and I think this is a weak argument in particular for sentry, where log messages get aggregated anyway on ingestion. I don't think the loss in readability is worth the extra rule and the cognitive extra work you got to do when reading the code. I'm -0.5 on that suggestion.